File: Dart/189.rotate_array.dart
void main() {
  rotate([1, 2, 3, 4, 5, 6, 7], 3);
  rotate([-1, -100, 3, 99], 2);
  rotate([1, 2], 3);
}

void rotate(List<int> nums, int k) {
  final list = <int>[];
  final offset = nums.length - k % nums.length;

  for (int i = 0; i < nums.length; i++) {
    list.add(nums[(i + offset) % (nums.length)]);
  }

  var index = 0;
  list.forEach((e) {
    nums[index++] = e;
  });
}

File: src/189.rotate_array.kt
fun main() {
    rotate(intArrayOf(1, 2, 3, 4, 5, 6, 7), 3)
    rotate(intArrayOf(-1, -100, 3, 99), 2)
    rotate(intArrayOf(1, 2), 3)
}

fun rotate(nums: IntArray, k: Int): Unit {
    val list = mutableListOf<Int>()
    val offset = nums.size - k % nums.size

    for (i in nums.indices) {
        list.add(nums[(i + offset) % (nums.size)])
    }

    var index = 0
    list.forEach { e ->
        nums[index++] = e
    }
}